backend/app/service/website.go

```diff
@@ -1343,6 +1343,14 @@ func (w WebsiteService) ChangePHPVersion(req request.WebsitePHPVersionReq) error
 	if runtime.Resource == constant.ResourceLocal || oldRuntime.Resource == constant.ResourceLocal {
 		return buserr.New("ErrPHPResource")
 	}
+	client, err := docker.NewDockerClient()
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+	if !checkImageExist(client, runtime.Image) {
+		return buserr.WithName("ErrImageNotExist", runtime.Name)
+	}
 	appInstall, err := appInstallRepo.GetFirst(commonRepo.WithByID(website.AppInstallID))
 	if err != nil {
 		return err
```

Comment:
There does not appear to be any obvious errors or inefficiencies in this code block, so no further information can be provided beyond a general recommendation for good practice:

No irregularities detected.

This is based on an understanding of current best practices and design principles relevant at the time they were written (September 2021), which may differ from more recent versions due to changing technologies, user needs, system requirements, security considerations, etc. For comprehensive analysis including performance tuning, it's recommended to review the latest standards and use modern tools like linters and analyzers, such as golangci-lint, go test coveragerc, go static struct checker, etc.

For example:

```go
// Ensure all dependencies are up-to-date with version checks

for _, dep := range pkg.Deps {
	if !isDepUpToDate(dep) {
		fmt.Println("Dependency", dep.ID, "not up to date.")
		go fixOldDependencies() // Optionally handle dependency resolution here...
		break
	}
}

if len(pkg.Errors) > 0 {
	log.Fatal(pkg.Errors...)
}

func isDepUpToDate(dep Dependency) bool {
   ...
   return true/false if condition else true/
}
```

Please note that these examples require some familiarity with Go programming language standards and context-specific conventions.